systemtest/cmd/apmbench/main.go

```diff
@@ -110,6 +113,33 @@ func benchmarkAgent(b *testing.B, l *rate.Limiter, expr string) {
 	})
 }
 
+func Benchmark10000AggregationGroups(b *testing.B, l *rate.Limiter) {
+	// Benchmark memory usage on aggregating high cardinality data.
+	// This should generate a lot of groups for service transaction metrics,
+	// transaction metrics, and service destination metrics.
+	//
+	// Using b.N instead of b.RunParallel since this benchmark is about memory
+	// usage.
+	for n := 0; n < b.N; n++ {
+		tracer := benchtest.NewTracer(b)
+		for i := 0; i < 10000; i++ {
+			if err := l.Wait(context.Background()); err != nil {
+				b.Fatal(err)
+			}
```

Comment:
An observation, since aggregations are time-based the rate limiter will mean that if the configured rate for the benchmark is less than 10_000 events per minute then the aggregation for a 1-minute window will split and the memory usage will be lower.

Reply:
It is true and that's what expected for the rate limiter to do. I'll add an comment to remind everyone on that.